flytekit/types/structured/structured_dataset.py

```diff
@@ -326,12 +326,13 @@ class StructuredDatasetTransformerEngine(TypeTransformer[StructuredDataset]):
         str: type_models.LiteralType(simple=type_models.SimpleType.STRING),
     }
 
-    ENCODERS: Dict[Type, Dict[str, Dict[str, StructuredDatasetEncoder]]] = {}
-    DECODERS: Dict[Type, Dict[str, Dict[str, StructuredDatasetDecoder]]] = {}
+    ENCODERS: Dict[Type, Dict[str, Dict[str, Handlers]]] = {}
```

Comment:
I feel like using `Handlers` is not correct here because the type of the class in the `ENCODERS` must be `StructuredDatasetEncoder`

Reply:
mypy making things painful 😄